coverage/control.py

```diff
@@ -653,7 +653,7 @@ def _atexit(self, event: str = "atexit") -> None:
             self._debug.write(f"{event}: pid: {os.getpid()}, instance: {self!r}")
         if self._started:
             self.stop()
-        if self._auto_save:
+        if self._auto_save or event == "sigterm":
```

Comment:
I'm curious why `_auto_save` isn't true, but I've only just taken a quick look at this pull request.

Reply:
Yeah I wondered that at first - it's being set by the `auto_data` arg to the `Coverage` class, which defaults to `False` and is passed as `True` by `multiproc.ProcessWithCoverage`, i.e. only in the multiprocessing case.

Reply:
That said, I'm more than happy with solving this another way - more than anything this proposed change serves as an illustration for what's going wrong!

Reply:
I'm trying to think this through.  On the one hand it seems odd: if `_auto_save` is false, then we won't save the data if the process ends normally, but we will if it's terminated?  But on the other, `sigterm=true` is documented as "to capture data at unusual process ends," which this definitely is. 

Maybe the answer is that `_auto_save` is only needed for the subprocess case.  For main processes, something else will be handling the data saving at normal process termination, and this is fine to put here.

I like that you have a test demonstrating the issue.  `event == "sigterm"` will only happen if `sigterm=true` is in the settings, so clearly the user wanted sigterm handling. 

Let's merge it!